src/app/components/chat/chat.component.ts

```diff
@@ -1,12 +1,82 @@
-import { Component } from '@angular/core';
+import { Component, OnInit, inject } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { ChatState } from '../../store/chats-store/chats.reducer';
+import { Store } from '@ngrx/store';
+import {
+  getOldMessages,
+  handleNewMessage,
+  sendNewMessage,
+} from '../../store/chats-store/chats.actions';
+import { selectMessages } from '../../store/chats-store/chats.selectors';
+import MessageInterface from '../../interfaces/chats';
+import { ChatService } from '../../services/chat.service';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
 
 @Component({
   selector: 'app-chat',
   standalone: true,
-  imports: [],
+  imports: [CommonModule, ReactiveFormsModule],
   templateUrl: './chat.component.html',
-  styleUrl: './chat.component.scss'
+  styleUrl: './chat.component.scss',
 })
-export class ChatComponent {
+export class ChatComponent implements OnInit {
+  chatStore: Store<ChatState> = inject(Store);
+  messages$ = this.chatStore.select(selectMessages);
 
+  chatService: ChatService = inject(ChatService);
+  socket: WebSocket | undefined;
+
+  clientID = crypto.randomUUID().toString();
+
+  messageForm = new FormGroup({
+    message: new FormControl(''),
+  });
+
+  ngOnInit() {
+    this.chatService.socket$.subscribe((socket) => {
+      this.socket = socket;
+    });
+    this.chatStore.dispatch(getOldMessages());
+    if (this.socket) {
```

Comment:
You can do all of this in the service and take all the logic out of this component